src/symbolic-node.jl

```diff
@@ -159,15 +160,19 @@ function set_attr(self :: SymbolicNode, key :: Symbol, value :: AbstractString)
 end
 
 #=doc
-.. function:: Variable(name :: Union{Base.Symbol, AbstractString})
+.. function:: Variable(name :: Union{Symbol, AbstractString})
 
    Create a symbolic variable with the given name. This is typically used as a placeholder.
    For example, the data node, acting as the starting point of a network architecture.
 =#
-function Variable(name :: Union{Base.Symbol, AbstractString})
+function Variable(name :: Union{Symbol, AbstractString}; attrs = Dict{Symbol, AbstractString}())
```

Comment:
Julia types not being covariant bite me a lot. This interface might be somehow inconvenient to use, as if the user write
```
attrs = Dict(:foo => "bar", :bar => "baz")
```
they will get a `Dict{Symbol,ASCIIString}` (or `UTF8String` depending on whether they used unicode), yet that is not a subtype of `Dict{Symbol,AbstractString}`, and the call will fail with a signature mismatch. 

A workaround that I'm currently using is in the signature, use general `attrs::Dict = Dict()` type, and then in the function body, do a explicit call to
```
attrs = convert(Dict{Symbol,AbstractString}, attrs)
```
attempt to get the right behavior and fail gracefully with meaningful error message. Can you modify it to be of this way? Thanks!